pkg/repo/migrations/helpers.go

```diff
@@ -0,0 +1,93 @@
+package migrations
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config"
+	"github.com/bacalhau-project/bacalhau/pkg/config/types"
+	"github.com/bacalhau-project/bacalhau/pkg/repo"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/spf13/viper"
+)
+
+func getViper(r repo.FsRepo) (*viper.Viper, error) {
+	repoPath, err := r.Path()
+	if err != nil {
+		return nil, err
+	}
+
+	configFile := filepath.Join(repoPath, config.ConfigFileName)
+	v := viper.New()
+	v.SetTypeByDefaultValue(true)
+	v.SetConfigFile(configFile)
+
+	// read existing config file if it exists
+	if err := v.ReadInConfig(); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
+	}
+	return v, nil
+}
+
+func configExists(r repo.FsRepo) (bool, error) {
+	repoPath, err := r.Path()
+	if err != nil {
+		return false, err
+	}
+
+	configFile := filepath.Join(repoPath, config.ConfigFileName)
+	_, err = os.Stat(configFile)
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return err == nil, err
+}
+
+func readConfig(r repo.FsRepo) (*viper.Viper, types.BacalhauConfig, error) {
+	v, err := getViper(r)
+	if err != nil {
+		return nil, types.BacalhauConfig{}, err
+	}
+	var fileCfg types.BacalhauConfig
+	if err := v.Unmarshal(&fileCfg, config.DecoderHook); err != nil {
+		return v, types.BacalhauConfig{}, fmt.Errorf("failed to unmarshal config file: %w", err)
+	}
+	return v, fileCfg, nil
+}
+
+// haveSameElements returns true if arr1 and arr2 have the same elements, false otherwise.
+func haveSameElements(arr1, arr2 []string) bool {
```

Comment:
Could move this to `pkg/lib` as `EqualSlices`

Reply:
I've copied this from the existing migration code. If I understand correctly, it checks if two _string_ slices are the same regardless of the order, as if order was also required, then `reflect.DeepEqual` should work. Not sure if the current implementation fits under `lib` as we will need to support generics or different types of slices for it to be useful as a lib. If you think it is still useful, then I can add it under `lib/collections` pkg

Reply:
Yeah or under an assertions package, its basically a copy of this: https://github.com/stretchr/testify/blob/v1.8.4/assert/assertions.go#L1039